### PR TITLE
fix: incorrect links in nav tree

### PR DESCRIPTION
**Pull request by RomainMuller**
When a markdown heading includes in-line links (`[label](href)` or
`[label][ref]`), the label of the item in the nav tree was incorectly
rendering raw markdown, and the anchor ID was not correct.

This addresses both of these issues:

- The anchor ID is now computed based on the recursive text elements
  from the DOM node (previously, it would not include any non-text
  child).

- The nav tree title now removes the markdown link decorations, and
  leaves only the label in the title & id.

- The nav tree title now removes custom anchors (`<a name="...">`)
  and leaves only the relevant text to compute the title & id.

Fixes https://github.com/cdklabs/construct-hub-webapp/issues/670

### CODE REVIEW

**Comment by RomainMuller** [comment]
BTW - not too sure how to test this... Pointers/ideas welcome!

**Comment by gabewomble** [comment]
> BTW - not too sure how to test this... Pointers/ideas welcome!

I believe there is a suite of tests regarding markdown parsing @MrArnoldPalmer wrote, I think that would be a good place to add tests for this!

**Comment by gabewomble** [comment]
It looks like nav items are still broken for `iam-floyd`, not sure if that case was intended to be solved with this PR, but for the case you linked, the changes work as expected!

**Comment by RomainMuller** [comment]
> It looks like nav items are still broken for `iam-floyd`

This one is also fixed now... It was a different cause.

File: src/views/Package/util.ts [hunk 1]
```diff
@@ -83,8 +83,17 @@ const getHeaderAttributes = (hdr: string): { id: string; title: string } => {
...
-  const [_, rawTitle] = /^#*\s*([^<]+?)\s*(?:<|$)/.exec(hdr) ?? [];
-  const wEmoji = rawTitle.replace(
+  const [_, rawTitle] =
+    /^#*\s*(?:<a\s.*?(?:\/>|<\/a>))?\s*([^<]+?)\s*(?:<|$)/.exec(hdr) ?? [
+      hdr,
+      hdr,
+    ];
+  const noLinkTitle = rawTitle.replace(
+    // Matches [label](text) and [label][text], replaces with just label.
+    /\[([^\]]+)\](?:\([^)]+\)|\[[^\]]+\])/g,
```
Comment (gabewomble) [inline src/views/Package/util.ts:93]: I'm convinced you dream in RegExp 😛

**Review by gabewomble — APPROVED** [review]
Nicely done 😎

File: src/views/Package/util.ts (hunk 1 above)
Comment (RomainMuller) [reply src/views/Package/util.ts:93]: I dream electric sheep.

**Pull request merged**